client/webserver/http.go

```diff
@@ -198,6 +201,59 @@ func (s *WebServer) handleWalletLogFile(w http.ResponseWriter, r *http.Request)
 	}
 }
 
+// handleGenerateQRCode is the handler for the '/generateqrcode' page request
+func (s *WebServer) handleGenerateQRCode(w http.ResponseWriter, r *http.Request) {
+	err := r.ParseForm()
+	if err != nil {
+		log.Errorf("error parsing form for generate qr code: %v", err)
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	address := r.Form["address"]
+	if len(address) != 1 || len(address[0]) == 0 {
+		log.Error("form for generating qr code does not contain address")
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	var darkMode bool
+	dark := r.Form["dark"]
+	if len(dark) >= 1 && len(dark[0]) > 0 {
+		darkMode, _ = strconv.ParseBool(dark[0])
+	}
+
+	qr, err := qrcode.New(address[0], qrcode.Medium)
+	if err != nil {
+		log.Error("error generating qr code: %v", err)
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	if darkMode {
+		qr.BackgroundColor = color.Black
+		qr.ForegroundColor = color.White
+	} else {
+		qr.BackgroundColor = color.White
+		qr.ForegroundColor = color.Black
+	}
```

Comment:
I have tried this with two scanners, and one is fine, but one cannot pick up the white qr code on black background. Can they always be white background with black foreground?

Reply:
Well, I tried both we on two scanners and they work well. Maybe scanner specific issue?

Reply:
They both work for me too. If it turns out to be a more widespread issue, we can have the light mode version always displayed by using a white/light patch behind it.

Reply:
They both work fine for me too. In the QR code library's docs there was an example with a black background and white foreground so I thought it would be fine, but now I'm doing some research and it's recommended that the foreground should be darker than the background.